src/site.rs

```diff
@@ -33,24 +35,41 @@ impl Site {
         self.options.is_valid()
     }
 
-    pub fn to_file_base_name(&self) -> String {
-        self.options
-            .filename
-            .clone()
-            .unwrap_or_else(|| filename::basename(&self.url))
+    pub fn to_file_path(&self) -> PathBuf {
+        self.options.filename.clone().unwrap_or_else(|| {
+            let folder = filename::domainfolder(&self.url);
+            let [first, rest @ ..] = &folder[..] else {
+                panic!(
+                    "domain has to have at least one segment {folder:?} {:?}",
+                    self.url
+                );
+            };
+            let mut path = Path::new(first).to_path_buf();
+            for f in rest {
+                path = path.join(f);
+            }
+            path.join(filename::filename(&self.url))
+        })
+    }
+
+    fn unique_idenfier(&self) -> String {
+        self.to_file_path()
+            .to_str()
+            .expect("the path is unicode already")
+            .to_string()
     }
 
-    pub fn get_all_file_basenames(sites: &[Self]) -> Vec<String> {
-        sites.iter().map(Self::to_file_base_name).collect()
+    pub fn get_all_file_paths(sites: &[Self]) -> Vec<PathBuf> {
+        sites.iter().map(Self::to_file_path).collect()
     }
 
     pub fn validate_no_duplicate(sites: &[Self]) -> Result<(), String> {
         // TODO: return url or something of specific duplicates
-        let mut file_basenames = Self::get_all_file_basenames(sites);
-        file_basenames.sort_unstable();
-        let total = file_basenames.len();
-        file_basenames.dedup();
-        if file_basenames.len() == total {
+        let mut uniq = sites.iter().map(Self::unique_idenfier).collect::<Vec<_>>();
+        uniq.sort_unstable();
```

Comment:
What's the reason for sorting?

Reply:
[`dedup` only works when sorted.](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup)

Reply:
I would add that as a comment to the code but I could see if you rather not.